gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -592,6 +593,41 @@ ChannelCredentials createS2ASecuredChannelCredentials() {
     }
   }
 
+  boolean isMtlsS2AHardBoundTokensEnabled() {
+    if (!useS2A) {
+      // If S2A cannot be used, {@code HardBoundTokenTypes.MTLS_S2A} hard bound tokens should not be
+      // used
+      return false;
+    }
+    if (allowedHardBoundTokenTypes == null
```

Comment:
nit: Can we also merge these fail-fast if checks together?

Reply:
Yes, done in https://github.com/googleapis/sdk-platform-java/pull/3591/commits/7c20422a57882b0243b298ef8e6f33293e4aa61a